typescript/libs/editor_client.py

```diff
@@ -68,11 +76,28 @@ def initialize(self):
             'typescript_auto_indent',
             'auto_match_enabled'
         ]:
-            settings.add_on_change(setting_name, self.load_format_settings)
+            settings.add_on_change(format_setting_name, self.load_format_settings)
         self.load_format_settings()
 
+        # load telemetry settings and set callbacks for setting changes
+        for other_setting_name in [
+            telemetry_setting_name,
+            'check_for_dts_updates'
+        ]:
+            settings.add_on_change(other_setting_name, self.load_other_settings)
+        self.load_other_settings()
```

Comment:
No need to load two sets of settings separately, which make `set_features` guaranteed to be called twice. One `load_settings` that called after all settings are read should be enough.